qutip/tests/test_floquet.py

```diff
@@ -32,24 +32,267 @@
+        output1 = fmmesolve(
+                            H, psi0, tlist, [c_op_fmmesolve], [],
+                            [noise_spectrum], T, args, floquet_basis=True)
+        # Calculate expectation values in the computational basis
+        p_ex = np.zeros(np.shape(tlist), dtype=complex)
+        for idx, t in enumerate(tlist):
+            f_modes_t = floquet_modes_t_lookup(f_modes_table_t, t, T)
+            f_states_t = [np.exp(-1j*t*f_energies[0])*f_modes_t[0],
+                          np.exp(-1j*t*f_energies[1])*f_modes_t[1]]
+            p_ex[idx] = expect(num(2), output1.states[idx].transform(
+                                f_states_t,
+                                True))
+
+        # Compare with mesolve
+        output2 = mesolve(H, psi0, tlist, c_op_mesolve, [], args)
+        p_ex_ref = expect(num(2), output2.states)
+
+        assert_(max(abs(np.real(p_ex)-np.real(p_ex_ref))) < 1e-4)
+
+    def testFloquetRates(self):
+        """
+        Compare rate transition and frequency transitions to analytical
+        results for a driven two-level system, for different drive amplitudes.
+        """
+
+        # Parameters
+        wq = 5.4 * 2 * np.pi
+        wd = 6.7 * 2 * np.pi
+        delta = wq - wd
+        T = 2 * np.pi / wd
+        tlist = np.linspace(0.0, 2 * T, 101)
+        array_A = np.linspace(0.001, 3 * 2 * np.pi, 10, endpoint=False)
+        H0 = Qobj(wq/2 * sigmaz())
+        arg = {'wd': wd}
+        c_ops = sigmax()
+        gamma1 = 1
+
+        delta_ana_deltas = []
+        array_ana_E0 = [-np.sqrt((delta/2)**2 + a**2) for a in array_A]
+        array_ana_E1 = [np.sqrt((delta/2)**2 + a**2) for a in array_A]
+        array_ana_delta = [2*np.sqrt((delta/2)**2 + a**2) for a in array_A]
+
+        def noise_spectrum(omega):
+            if omega > 0:
+                return 0.5 * gamma1 * omega/(2*np.pi)
+            else:
+                return 0
+
+        idx = 0
+        for a in array_A:
+            # Hamiltonian
+            H1_p = Qobj(a * sigmap())
+            H1_m = Qobj(a * sigmam())
+            H = [H0, [H1_p, lambda t, args: np.exp(-1j * arg['wd'] * t)],
+                     [H1_m, lambda t, args: np.exp(1j * arg['wd'] * t)]]
+
+            # Floquet modes
+            fmodes0, fenergies = floquet_modes(H, T, args={}, sort=True)
+            f_modes_table_t = floquet_modes_table(fmodes0, fenergies,
+                                                  tlist, H, T,
+                                                  args={})
+            # Get X delta
+            DeltaMatrix, X, frates, Amat = floquet_master_equation_rates(
+                                            fmodes0, fenergies,
+                                            c_ops, H, T, {},
+                                            noise_spectrum, 0, 5)
+            # Check energies
+            deltas = np.ndarray.flatten(DeltaMatrix)
+
+            assert_(min(abs(deltas-array_ana_delta[idx])) < 1e-4)
```

Comment:
`min` doesn't seem right?  That succeeds as long as any pair of elements is close, rather than requiring _all_ pairs to be close.  If it _is_ meant to be the latter, it should again be `np.testing.assert_allclose`.